### PR TITLE
Contract Provider

### PR DESCRIPTION
**Pull request by JamesDalboth**
This removes that annoying passing down of contract and accounts now that we have appropriate providers to retrieve those values instead. This required some classes to be converted into React.FC's which is always good when it happens.

### CODE REVIEW

File: app/src/components/RegisterArtist.tsx [hunk 1]
```diff
@@ -75,40 +78,35 @@ class RegisterArtist extends React.Component<ContractProps, RegisterArtistState>
...
 
-    this.setState({ validated: true, submitted: true });
-
-    const { contracts, accounts } = this.props;
+    setValidated(true);
+    setSubmitted(true);
 
     // eslint-disable-next-line
-    const { metaIpfsHash, ...restOfTheFields } = this.state.fields;
+      const { metaIpfsHash, ...restOfTheFields } = fields;
```
Comment (DJRHails) [inline app/src/components/RegisterArtist.tsx]: Accidental spacing

File: app/src/components/ArtworkInfo.tsx [hunk 2]
```diff
@@ -41,102 +41,89 @@ interface ArtworkInfoState {
...
+        {children}
+      </ArtworkCard>
+    );
+  } else {
+    return (
+      <ArtworkCard img={imgSrc} fullscreen={fullscreen}>
+        {children}
+      </ArtworkCard>
+    );
+  };
```
Comment (DJRHails) [inline app/src/components/ArtworkInfo.tsx]: -10,000 marks

**Review by DJRHails — APPROVED** [review]

File: app/src/components/ArtworkInfo.tsx (hunk 2 above)
Comment (JamesDalboth) [reply app/src/components/ArtworkInfo.tsx]: :(

Comment (JamesDalboth) [reply app/src/components/ArtworkInfo.tsx]: tbf according to git blame neil wrote the -10000 marks bit a month ago we so it's an old bug

**Pull request merged**